sdk_mock.go

```diff
@@ -0,0 +1,141 @@
+// Code generated by MockGen. DO NOT EDIT.
+// Source: sdk.go
+
+// Package sdk is a generated GoMock package.
+package sdk
+
+import (
+	gomock "github.com/golang/mock/gomock"
+	reflect "reflect"
+)
+
+// MockRPCClient is a mock of RPCClient interface
+type MockRPCClient struct {
+	ctrl     *gomock.Controller
+	recorder *MockRPCClientMockRecorder
+}
+
+// MockRPCClientMockRecorder is the mock recorder for MockRPCClient
+type MockRPCClientMockRecorder struct {
+	mock *MockRPCClient
+}
+
+// NewMockRPCClient creates a new mock instance
+func NewMockRPCClient(ctrl *gomock.Controller) *MockRPCClient {
+	mock := &MockRPCClient{ctrl: ctrl}
+	mock.recorder = &MockRPCClientMockRecorder{mock}
+	return mock
+}
+
+// EXPECT returns an object that allows the caller to indicate expected use
+func (m *MockRPCClient) EXPECT() *MockRPCClientMockRecorder {
+	return m.recorder
+}
+
+// Call mocks base method
+func (m *MockRPCClient) Call(result interface{}, method string, args ...interface{}) error {
+	varargs := []interface{}{result, method}
+	for _, a := range args {
+		varargs = append(varargs, a)
+	}
+	ret := m.ctrl.Call(m, "Call", varargs...)
+	ret0, _ := ret[0].(error)
+	return ret0
+}
+
+// Call indicates an expected call of Call
+func (mr *MockRPCClientMockRecorder) Call(result, method interface{}, args ...interface{}) *gomock.Call {
+	varargs := append([]interface{}{result, method}, args...)
+	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "Call", reflect.TypeOf((*MockRPCClient)(nil).Call), varargs...)
+}
+
+// MockSDK is a mock of SDK interface
+type MockSDK struct {
+	ctrl     *gomock.Controller
+	recorder *MockSDKMockRecorder
+}
+
+// MockSDKMockRecorder is the mock recorder for MockSDK
+type MockSDKMockRecorder struct {
+	mock *MockSDK
+}
+
+// NewMockSDK creates a new mock instance
+func NewMockSDK(ctrl *gomock.Controller) *MockSDK {
+	mock := &MockSDK{ctrl: ctrl}
+	mock.recorder = &MockSDKMockRecorder{mock}
+	return mock
+}
+
+// EXPECT returns an object that allows the caller to indicate expected use
+func (m *MockSDK) EXPECT() *MockSDKMockRecorder {
+	return m.recorder
+}
+
+// Login mocks base method
```

Comment:
As mentioned before we don't need all those methods on the mock, so better don't define until we need them :)